File: internal/content/testdata/bench/csharp/nc-right-side-view.cs
public class Solution {
    public long[] right_side_view(TreeNode root) {
        if (root == null) return new long[0];
        var q = new Queue<TreeNode>();
        var res = new List<long>();
        q.Enqueue(root);
        while (q.Count > 0) {
            int n = q.Count;
            for (int i = 0; i < n; i++) {
                TreeNode node = q.Dequeue();
                if (i == n - 1) res.Add(node.val);
                if (node.left != null) q.Enqueue(node.left);
                if (node.right != null) q.Enqueue(node.right);
            }
        }
        return res.ToArray();
    }
}

File: internal/content/testdata/bench/go/nc-right-side-view.go
func right_side_view(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}
	result := []int{}
	queue := []*TreeNode{root}
	for len(queue) > 0 {
		n := len(queue)
		for i := 0; i < n; i++ {
			node := queue[i]
			if i == n-1 {
				result = append(result, node.Val)
			}
			if node.Left != nil {
				queue = append(queue, node.Left)
			}
			if node.Right != nil {
				queue = append(queue, node.Right)
			}
		}
		queue = queue[n:]
	}
	return result
}
